code/datums/components/battery_power.dm

```diff
@@ -0,0 +1,51 @@
+/// Generalized framework allowing something to draw power from an inserted power cell.
+/// Contains attack_hand() and attackby() signals allowing cells to be added and removed by hand.
+/// This base type supports only standard cells; device cells won't fit.
+/datum/component/battery_power
+	var/obj/item/cell/cell
+
+/datum/component/battery_power/Initialize()
+	if (istype(parent, /obj/machinery))
```

Comment:
I suspect we'll eventually want to do something about this for not-machinery types.

Reply:
yeye; I figured that the component could be expanded in the future, at which point this kinda thing could be amended